src/app/core/admin/admin-dashboard/admin-dashboard.component.ts

```diff
@@ -0,0 +1,209 @@
+import { FormsModule } from "@angular/forms";
+import { MatButtonModule } from "@angular/material/button";
+import { MatCardModule } from "@angular/material/card";
+import { MatTooltipModule } from "@angular/material/tooltip";
+import { MatIconModule } from "@angular/material/icon";
+import { MatDialog } from "@angular/material/dialog";
+import { firstValueFrom } from "rxjs";
+import {
+  AdminWidgetDialogComponent,
+  AdminWidgetDialogData,
+} from "../admin-widget-dialog/admin-widget-dialog.component";
+import { ViewTitleComponent } from "../../common-components/view-title/view-title.component";
+import { WidgetComponentSelectComponent } from "../admin-entity-details/widget-component-select/widget-component-select.component";
+import { MatSnackBar } from "@angular/material/snack-bar";
+
+@Component({
+  selector: "app-admin-dashboard",
+  standalone: true,
+  imports: [
+    DynamicComponentDirective,
+    DragDropModule,
+    FaIconComponent,
+    FormsModule,
+    MatButtonModule,
+    MatCardModule,
+    MatTooltipModule,
+    MatIconModule,
+    ViewTitleComponent,
+  ],
+  templateUrl: "./admin-dashboard.component.html",
+  styleUrls: [
+    "./admin-dashboard.component.scss",
+    "../../dashboard/dashboard/dashboard.component.scss",
+    "../admin-entity/admin-entity-styles.scss",
+  ],
+})
+export class AdminDashboardComponent implements OnInit {
+  @Input() dashboardViewId: string;
+  @Input() isDisabled: boolean = false;
+
+  dashboardConfig: DashboardConfig;
+  private originalDashboardConfig: DashboardConfig;
+
+  private readonly configService = inject(ConfigService);
+  private readonly dialog = inject(MatDialog);
+  private readonly location = inject(Location);
+  private readonly snackBar = inject(MatSnackBar);
+
+  ngOnInit() {
+    this.loadDashboardViewConfig();
+  }
+
+  drop(event: CdkDragDrop<any[]>) {
+    if (event.previousContainer === event.container) {
+      moveItemInArray(
+        this.dashboardConfig.widgets,
+        event.previousIndex,
+        event.currentIndex,
+      );
+    } else {
+      // Handle drop from available widgets
+      const widgetType = event.previousContainer.data[event.previousIndex];
+      const newWidget: DynamicComponentConfig = {
+        component: widgetType.component,
+        config: { ...widgetType.defaultConfig },
+      };
+
+      this.dashboardConfig.widgets.splice(event.currentIndex, 0, newWidget);
+    }
+  }
+
+  async editWidget(widgetConfig: DynamicComponentConfig, idx: number) {
+    const settingsComponent = this.getSettingsComponentforWidget(widgetConfig);
+    if (!settingsComponent) return;
+
+    const updatedConfig = await this.openWidgetSettingsDialog(
+      widgetConfig,
+      settingsComponent,
+    );
+    if (updatedConfig) {
+      this.dashboardConfig.widgets[idx] = updatedConfig;
+      this.dashboardConfig.widgets = [...this.dashboardConfig.widgets];
+    }
+  }
+
+  getSettingsComponentforWidget(widgetConfig: DynamicComponentConfig): any {
```

Comment:
we can try to design a more generic way to register dashboard components, so that this hard-coded list is not necessary 

[--> Pair Programming Session]